src/main/java/com/google/finapp/SpannerDaoJDBCImpl.java

```diff
@@ -0,0 +1,197 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.finapp;
+
+import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.*;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Arrays;
+
+final class SpannerDaoJDBCImpl implements SpannerDaoInterface {
```

Comment:
Do we plan on adding a flag and switching the implementations based on that in https://github.com/GoogleCloudPlatform/cloud-spanner-samples/blob/main/src/main/java/com/google/finapp/DatabaseModule.java

Also please test this implementation out locally.

Reply:
Will be handled with https://github.com/GoogleCloudPlatform/cloud-spanner-samples/issues/8.

This implementation has been tested (at least with the given RPCs) locally.